lib/matplotlib/testing/decorators.py

```diff
@@ -393,11 +397,23 @@ def image_comparison(baseline_images, extensions=None, tol=0,
         Optional name for the base style to apply to the image test. The test
         itself can also apply additional styles if desired. Defaults to the
         '_classic_test' style.
-
     """
+
     if extensions is None:
         # default extensions to test
         extensions = ['png', 'pdf', 'svg']
+        if baseline_images is not None:
+            baseline_exts = [*filter(None, {Path(baseline).suffix[1:]
+                                            for baseline in baseline_images})]
+            if baseline_exts:
+                if len(baseline_exts) > 1:
+                    raise ValueError(
+                        "When including extensions directly in "
+                        "'baseline_images', all baselines must share the same "
+                        "suffix.")
+                extensions = [*baseline_exts]
```

Comment:
Why do you need a copy?

Reply:
oops, that came from an old version, removed the copy.